conjure-java-core/src/test/resources/ete-service.yml

```diff
@@ -223,3 +223,13 @@ services:
           ints:
             type: set<integer>
             param-type: query
+
+      receiveListOfOptionals:
+        http: PUT /list/optionals
+        args:
+          value: list<optional<string>>
+
+      receiveSetOfOptionals:
+        http: PUT /set/optionals
+        args:
+          value: set<optional<string>>
```

Comment:
is it worth also validating the case where you receive a raw map? like `map<string,string>`, I guess we probably want to reject `{"hello": null}`?

Reply:
I think so, but I didn't include it in this change because immutablemap will additionally validate against duplicate keys. I'd prefer to implement that separately.